custom/icds_reports/tests/agg_tests/test_governance_apis.py

```diff
@@ -83,3 +83,60 @@ def test_data_fetching_state_names_api(self):
         data = get_state_names()
         expected_count = 6
         self.assertEqual(len(data), expected_count)
+
+    def test_data_fetching_total_record_count_for_vhnds_visit_api(self):
+        """
+        test to check the total count of records that are returned from the vhnds api
+        """
+        limit = GOVERNANCE_API_HOME_VISIT_RECORDS_PAGINATION
+        query_filters = {'state_id': 'st1'}
+        order = ['state_name', 'district_name', 'block_name', 'supervisor_name', 'awc_name']
+        data, count = get_vhnd_data(limit,
+                                    2017, 5, order, query_filters)
+        expected_count = 10
+        self.assertEqual(count, expected_count)
+
+    def test_data_fetching_without_start_for_vhnds_api(self):
+        """
+        test to check the first record that is returned from the vhnds api without start parameter
+        """
+        limit = GOVERNANCE_API_HOME_VISIT_RECORDS_PAGINATION
+        query_filters = {'state_id': 'st1'}
+        order = ['state_name', 'district_name', 'block_name', 'supervisor_name', 'awc_name']
+        data, count = get_vhnd_data(limit,
+                                    2017, 5, order, query_filters)
+        expected_first_row = {
+            "state": "st1", "district": "d1", "block": "b1", "sector": "s1", "awc": "a41",
+            "month": datetime.date(2017, 5, 1), "vhsnd_conducted": "no", "vhsnd_date": "Data Not Entered",
+            "anm_present": "no", "asha_present": "no", "any_child_immunized": "no",
+            "anc_conducted": "no"
+        }
+        self.assertEqual(data[0], expected_first_row)
+
+    def test_data_fetching_with_start_for_vhnds_api(self):
+        """
+        test to check the first record that is returned from the vhnds api with start parameter
+        """
+        limit = GOVERNANCE_API_HOME_VISIT_RECORDS_PAGINATION
+        query_filters = {'state_id': 'st1', 'awc_id__gt': 'a41'}
+        order = ['state_name', 'district_name', 'block_name', 'supervisor_name', 'awc_name']
+        data, count = get_vhnd_data(limit, 2017, 5, order, query_filters)
+        expected_first_row = {
+            "state": "st1", "district": "d1", "block": "b1", "sector": "s1", "awc": "a49",
+            "month": datetime.date(2017, 5, 1), "vhsnd_conducted": "no", "vhsnd_date": "Data Not Entered",
+            "anm_present": "no", "asha_present": "no", "any_child_immunized": "no",
+            "anc_conducted": "no"
+        }
+        self.assertEqual(data[0], expected_first_row)
+
+    def test_data_fetching_total_record_count_for_no_records_for_vhnds_api(self):
```

Comment:
Can you also add a test case to test if there are multiple vhnds in a moth for one awc it is giving you the first one. That will also be be the validation of your aggregation code.

Reply:
added